### DISCUSSION TITLE
How does autotuner handle the case of no solution?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#1041] [General] [answered]
@nerriav I'm trying to optimize a design that will fail io floorplanning if some of the search parameters for autotuner have unfortunate combinations.

How does is the Autotuner designed to handle these cases?

In my trial run, I had 10 runs, but all failed io floorplanning, yet I get "Best parameters found".

**Comment by nerriav** [comment]
It seems that the default of the tool is to show the "best results message" when done running. 
same case when we run it on different location - didn't run at all but still gave this message.
From what I see in the script, no check is been done to validate the result (only check I see is about the input parameter file)

**Comment by maliberty** [comment, marked as answer]
Failed runs are assigned a very high score in evaluate().  You could check what the score of the best result is and if it is too large it must represent a failure.
